File: backend/app/schemas/document.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict

from app.models.document import DocumentStatus
from app.utils.enum import StorageProvider


class DocumentResponse(BaseModel):
    id: UUID
    workspace_id: UUID
    mime_type: str
    status: DocumentStatus
    storage_provider: StorageProvider
    error_message: str | None
    created_at: datetime
    uploaded_by: UUID

    model_config = ConfigDict(from_attributes=True)


class DocumentListResponse(BaseModel):
    items: list[DocumentResponse]
    total: int


class DocumentUploadResponse(BaseModel):
    document: DocumentResponse
    message: str
    

File: frontend/src/types/document.ts
export type DocumentStatus =
  | "pending"
  | "processing"
  | "completed"
  | "failed";

export type StorageProvider = "local" | "minio";

export interface DocumentItem {
  id: string;
  workspace_id: string;
  original_filename: string;
  mime_type: string;
  file_extension: string;
  file_size: number;
  checksum: string;
  status: DocumentStatus;
  storage_provider: StorageProvider;
  error_message: string | null;
  created_at: string;
  updated_at: string;
}

export interface DocumentUploadResponse {
  document: DocumentItem;
  message: string;
}

export interface DocumentListResponse {
  items: DocumentItem[];
  total: number;
}
